packages/form/addon/components/cf-field/input/files.js

```diff
@@ -0,0 +1,111 @@
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { macroCondition, isTesting } from "@embroider/macros";
+import Component from "@glimmer/component";
+import { queryManager } from "ember-apollo-client";
+import fetch from "fetch";
+
+import getFilesAnswerInfoQuery from "@projectcaluma/ember-form/gql/queries/filesanswer-info.graphql";
+
+export default class CfFieldInputFilesComponent extends Component {
+  @service intl;
+
+  @queryManager apollo;
+
+  get files() {
+    return this.args.field?.answer?.value;
+  }
+
+  @action
+  async download(fileId) {
+    if (!fileId) {
+      return;
+    }
+    const answers = await this.apollo.query(
+      {
+        query: getFilesAnswerInfoQuery,
+        variables: { id: this.args.field.answer.raw.id },
+        fetchPolicy: "network-only",
+      },
+      "node.value"
+    );
+    const { downloadUrl } =
+      answers.find((file) =>
+        macroCondition(isTesting())
+          ? file.id === fileId ||
+            atob(file.id).substring(file.__typename.length + 1) === fileId
+          : file.id === fileId
+      ) ?? {};
+    if (downloadUrl) {
+      window.open(downloadUrl, "_blank");
+    }
+  }
+
+  @action
+  async save({ target }) {
+    // store the old list of files
+    const fileList = [...(this.files || [])];
+
+    // unwrap files from FileList construct
+    const newFiles = Array.from(target.files).map((file) => ({
+      name: file.name,
+      value: file,
+    }));
+    newFiles.forEach((file) => fileList.push({ name: file.name }));
```

Comment:
```suggestion
   fileList = [...fileList, ...newFiles]
```
or if you really can't have the `id` property in there
```suggestion
   fileList = [...fileList, ...newFiles.map(({ name }) => ({ name })]
```

Reply:
First option works fine! :+1: